File: server/src/models/communication.rs
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};
use uuid::Uuid;

#[derive(Debug, Serialize)]
pub struct MailboxParty {
    pub name: String,
    pub email: String,
}

#[derive(Debug, Serialize)]
pub struct MailboxMessage {
    pub id: Uuid,
    pub from: MailboxParty,
    pub to: String,
    pub subject: String,
    pub snippet: String,
    pub body: String,
    pub sent_at: DateTime<Utc>,
    pub read: bool,
    pub starred: bool,
    pub folder: String,
    pub has_attachment: bool,
}

#[derive(Debug, Serialize)]
pub struct MailboxListResponse {
    pub messages: Vec<MailboxMessage>,
}

#[derive(Debug, Serialize)]
pub struct UnreadCountResponse {
    pub unread_inbox: i64,
}

#[derive(Debug, Deserialize)]
pub struct SendMessageRequest {
    pub to_email: Option<String>,
    pub subject: String,
    pub body: String,
    #[serde(default)]
    pub draft: bool,
}

#[derive(Debug, Deserialize)]
pub struct PatchMailboxRequest {
    pub read: Option<bool>,
    pub starred: Option<bool>,
    pub folder: Option<String>,
}

#[derive(Debug, Serialize)]
pub struct SendMessageResponse {
    pub id: Uuid,
}

File: server-new/internal/models/communication/communication.go
// Package communication holds JSON shapes for the mailbox API (server/src/models/communication.rs).
package communication

import (
	"time"

	"github.com/google/uuid"
)

// Party is a mailbox participant (name + email for display).
type Party struct {
	Name  string `json:"name"`
	Email string `json:"email"`
}

// MailboxMessage is a single message view for the authenticated user's folder.
type MailboxMessage struct {
	ID            uuid.UUID `json:"id"`
	From          Party     `json:"from"`
	To            string    `json:"to"`
	Subject       string    `json:"subject"`
	Snippet       string    `json:"snippet"`
	Body          string    `json:"body"`
	SentAt        time.Time `json:"sentAt"`
	Read          bool      `json:"read"`
	Starred       bool      `json:"starred"`
	Folder        string    `json:"folder"`
	HasAttachment bool      `json:"hasAttachment"`
}

// MailboxListResponse is returned by GET /api/v1/communication/messages?folder=...
type MailboxListResponse struct {
	Messages []MailboxMessage `json:"messages"`
}

// UnreadCountResponse is returned by GET /api/v1/communication/unread-count
type UnreadCountResponse struct {
	UnreadInbox int64 `json:"unreadInbox"`
}

// SendMessageRequest is the body for POST (send or draft).
type SendMessageRequest struct {
	ToEmail *string `json:"toEmail"`
	Subject string  `json:"subject"`
	Body    string  `json:"body"`
	Draft   bool    `json:"draft"`
}

// PatchMailboxRequest patches read/starred/folder for a mailbox entry.
type PatchMailboxRequest struct {
	Read    *bool   `json:"read"`
	Starred *bool   `json:"starred"`
	Folder  *string `json:"folder"`
}

// SendMessageResponse returns the new message id after send or save draft.
type SendMessageResponse struct {
	ID uuid.UUID `json:"id"`
}
